### DISCUSSION TITLE
Early alchemiscale results

### DISCUSSION THREAD

**Discussion by IAlibay** [openforcefield/discussions#24] [alchemiscale User Questions]
Since @dotsdl has both a poster at the free energy workshop and a talk at the OMSF workshop, we thought it would be good to aggregate initial results here.

**Comment by IAlibay** [comment]
Update on my end: we should have a full set of openff 2.0 and 2.1-rc1 comparisons for TYK2, Thrombin, and MCL-1 by ~ Thursday (I'm flying tomorrow so won't be able to aggregate).

Input data:
- For an initial "backwards compatibility" check, we're using the data from https://github.com/OpenFreeEnergy/openfe-benchmarks/tree/main/openfe_benchmarks/data (future iterations will use the PLB data directly).

Protocol details:
- 3 replicas
- HREX w/ 11 windows
- 2 ns equilibration and "up to" 5 ns production (early termination once the error drops below 0.2 kT)
- HMR (3 amu) + 4 fs timestep
- cutoff is set at 0.9 nm, we aren't doing unsampled edges

**Reply by IAlibay** [reply]
MLC1 results (will update with data later)
![image](https://user-images.githubusercontent.com/12460125/235702740-a22d0887-6636-4d5d-8f23-8eaf1bd29dc0.png)
[RBFE_mcl1_openfe_hrex_off2.0.csv](https://github.com/openforcefield/discussions/files/11453703/RBFE_mcl1_openfe_hrex_off2.0.csv)
[RBFE_mcl1_openfe_hrex_off2.1.csv](https://github.com/openforcefield/discussions/files/11453704/RBFE_mcl1_openfe_hrex_off2.1.csv)

**Reply by IAlibay** [reply]
TYK2 results (will update with data later)
![image](https://user-images.githubusercontent.com/12460125/235702880-3bc25528-0cf6-412e-b846-f8845742bb86.png)

[RBFE_tyk2_openfe_hrex_off2.0.csv](https://github.com/openforcefield/discussions/files/11453708/RBFE_tyk2_openfe_hrex_off2.0.csv)
[RBFE_tyk2_openfe_hrex_off2.1.csv](https://github.com/openforcefield/discussions/files/11453709/RBFE_tyk2_openfe_hrex_off2.1.csv)

**Reply by IAlibay** [reply]
Thrombin results (will update with data later)
![image](https://user-images.githubusercontent.com/12460125/235702952-de168bcd-2a85-42cb-8e82-e85d2aabb246.png)

[RBFE_thrombin_openfe_hrex_off2.0.csv](https://github.com/openforcefield/discussions/files/11453711/RBFE_thrombin_openfe_hrex_off2.0.csv)
[RBFE_thrombin_openfe_hrex_off2.1.csv](https://github.com/openforcefield/discussions/files/11453712/RBFE_thrombin_openfe_hrex_off2.1.csv)

**Reply by ijpulidos** [reply]
These look really good, thanks for sharing. Now the results for tyk2 look more similar to the ones we have with `perses` repex. Any ideas why the error bars on MCL1 seem higher for off2.1? I found that interesting.

**Reply by IAlibay** [reply]
@ijpulidos

re: repex vs nec - I'm going to update with the TYK2 comparison on your thread and keep this one for off2.0 vs off2.1.

re: mcl1 - honestly this is just a difficult target, it's got a wide enough binding site that ligands can easily tumble around. Especially given that I'm using early termination on the repex protocol, it's very likely we didn't sufficiently sample enough and it's just bad luck.

**Reply by IAlibay** [reply]
hif2a
![image](https://github.com/openforcefield/discussions/assets/12460125/3ad91a1a-3859-4a9e-a8cd-f5db314fff59)


[RBFE_hif2a_openfe_hrex_off2.0.csv](https://github.com/openforcefield/discussions/files/11453724/RBFE_hif2a_openfe_hrex_off2.0.csv)
[RBFE_hif2a_openfe_hrex_off2.1.csv](https://github.com/openforcefield/discussions/files/11453725/RBFE_hif2a_openfe_hrex_off2.1.csv)

**Reply by IAlibay** [reply]
p38

![image](https://github.com/openforcefield/discussions/assets/12460125/fe73574d-3ca4-4b01-8946-18be27c89380)

[RBFE_p38_openfe_hrex_off2.0.csv](https://github.com/openforcefield/discussions/files/11453727/RBFE_p38_openfe_hrex_off2.0.csv)
[RBFE_p38_openfe_hrex_off2.1.csv](https://github.com/openforcefield/discussions/files/11453728/RBFE_p38_openfe_hrex_off2.1.csv)

**Reply by IAlibay** [reply]
All results

![image](https://github.com/openforcefield/discussions/assets/12460125/a49c4ae1-99ec-4d12-9b18-7bb63767249c)

**Reply by dotsdl** [reply]
Awesome! Thank you for this @IAlibay!!!

**Reply by dotsdl** [reply]
@IAlibay any thoughts on the outliers on p38? in the DDG plot (right)?

**Reply by IAlibay** [reply]
@dotsdl see here: https://openforcefieldgroup.slack.com/archives/CKSHCE7SB/p1683813489278679?thread_ts=1683813489.278679&cid=CKSHCE7SB

Short answer is that we don't know right now (I don't have tools to know what the FF changes were for these two ligands).

**Reply by dotsdl** [reply]
Ah cool, just wanted to make sure I was able to answer the question if I get it tomorrow; much appreciated!

**Comment by ijpulidos** [comment]
These are the first results I obtained running the NEQ cycling protocol from perses. I'm uploading the figures and the `arsenic`/`cinnabar` csv file that I used for them.

These are results from the `tyk2` system in https://github.com/OpenFreeEnergy/openfe-benchmarks/tree/main/openfe_benchmarks/data 

![plot_absolute](https://user-images.githubusercontent.com/2949729/235229808-54fc6dd4-f6c0-436d-b69d-81d5740b787a.png)
![plot_relative](https://user-images.githubusercontent.com/2949729/235229811-2a621354-c587-49c0-8fb1-4ecf6d1cc660.png)
[tyk2_arsenic.csv](https://github.com/openforcefield/discussions/files/11356511/tyk2_arsenic.csv)

**Reply by dotsdl** [reply]
Awesome, thank you @ijpulidos! Can you give some commentary as to the overall picture this paints for us here? Is this what we would expect given the limitations of the NEQ cycling method?

**Reply by IAlibay** [reply]
@dotsdl - I've got the TYK2 REPEX results ready (at least one version of them), so I'm going to do the head-to-head comparison a bit later today or tomorrow. From what I can see the results are similar, except that there's one particluarly bad edge in the NEQ protocol.

**Reply by ijpulidos** [reply]
I would wait for @IAlibay results. I'm sharing the notebook I used to get these results and analyze the data here, please mind this is still in a very early stage and there is a lot of unnecessary things in the notebook. I basically created the `csv` file for arsenic/cinnabar manually fro my results (I can share some tools that help with that if needed) and run the relevant cells in the notebook (where you see the plots). 

[neq_cyc_tyk2_results.tar.gz](https://github.com/openforcefield/discussions/files/11367018/neq_cyc_tyk2_results.tar.gz)

**Reply by ijpulidos** [reply]
@dotsdl as to the overall picture, I have seen better results using repex with perses, though with different transformations. These were obtained with the data in `openfe-benchmarks` whereas other results I have have been obtained with the data in `protein-ligand-benchmarks` repo in https://github.com/openforcefield/protein-ligand-benchmark/tree/main/data/tyk2

**Reply by ijpulidos** [reply]
And to be clear, this was obtained form a neq cycling simulation using `250000` for both eq and neq steps, with a 4fs timestep. That means 1ns of simulation time for each of the eq and neq parts (considering it is a cycle, so each part is run twice).

**Reply by IAlibay** [reply]
Unfortunately had a couple of failures on one of my "no early termination" tyk2 runs, waiting on that to complete first (seems like there's a bit of a queue at the moment).

**Reply by IAlibay** [reply]
Ok, here's a comparison between HREX and NECycling (re-plotted because I changed the plotting style and I didn't want that to be the source of confusion):

![image](https://user-images.githubusercontent.com/12460125/235753614-5d1e919e-8328-4933-a553-d01e0631bd57.png)


The HREX results here use the same protocol details as those outlined here: https://github.com/orgs/openforcefield/discussions/24#discussioncomment-5722528

Note the 0.2 kT label represents the early termination threshold (i.e. it'll run for up to 11 ns per window, or until a 0.2 kT threshold is reached, whatever happens first).

It looks like all the problems seem to stem from just one edge: lig_ejm_31 -> lig_ejm_54, which is overestimated by the NECycling protocol.

Just to check that this wasn't a sampling issue, I reran the TYK2 system without turning on early termination (labelled as "11 * 7 ns"). Those are consistent.

<img src="https://user-images.githubusercontent.com/12460125/235755053-613adcf5-2a46-47a8-98d1-41008d280557.png" width="400" height="400">

We could try running the repex protocol much longer for the one edge, maybe it's some kind of slow loop motion?

**Reply by IAlibay** [reply]
@ijpulidos - is it possible that this used the old `gufe` default for hmass? I've seen bad edges on 4 amu mass HMR just randomly pop up with relative hydration free energies.

**Reply by dotsdl** [reply]
> We could try running the repex protocol much longer for the one edge, maybe it's some kind of slow loop motion?

@IAlibay did you mean here that we should try running longer with the neqcyc protocol, not repex, since we see the overestimation in the neqcyc result for that one edge?

**Reply by IAlibay** [reply]
@dotsdl - yeah sorry neqcyc should be redone in the first instance. But we should probably run both to exhaustion just to check tbh.

**Reply by ijpulidos** [reply]
> @ijpulidos - is it possible that this used the old `gufe` default for hmass? I've seen bad edges on 4 amu mass HMR just randomly pop up with relative hydration free energies.

I did use gufe 0.7.2 which is not the latest release. But I'm explicitely setting `settings.forcefield_settings.hydrogen_mass = 3.0`, so it should be using the correct value.

> @dotsdl - yeah sorry neqcyc should be redone in the first instance. But we should probably run both to exhaustion just to check tbh.

For now I think a quick way to check is re-running the "problematic" lig_ejm_31 -> lig_ejm_54 transformation for a longer time, I will double check the performance we are getting and increase the simulation times accordingly.

**Comment by ppxasjsm** [comment]
Wow these are really cool!
